Framework/API/src/ExperimentInfo.cpp

```diff
@@ -69,7 +69,7 @@ ExperimentInfo::ExperimentInfo()
       m_run(new Run()), m_parmap(new ParameterMap()),
       sptr_instrument(new Instrument()),
       m_detectorInfo(boost::make_shared<Beamline::DetectorInfo>()),
-      m_componentInfo(boost::make_shared<Beamline::ComponentInfo>()) {
+      m_componentInfo(boost::shared_ptr<Beamline::ComponentInfo>(nullptr)) {
```

Comment:
I was trying to avoid a default constructor for a `ComponentInfo`. But you're probably right here. I hadn't considered it if `setInstrument` was not called.